lib/watir/logger.rb

```diff
@@ -22,26 +22,36 @@ class Logger
 
     def_delegators :@logger, :debug, :debug?,
                    :info, :info?,
-                   :warn, :warn?,
+                   :warn?,
                    :error, :error?,
                    :fatal, :fatal?,
                    :level
 
-    def initialize
+    def initialize(progname="Watir")
       @logger = create_logger($stdout)
+      @logger.progname = progname
+      @ignored = []
+    end
+
+    def ignore(id)
+      @ignored << id
     end
 
     def output=(io)
-      # `Logger#reopen` was added in Ruby 2.3
-      if @logger.respond_to?(:reopen)
-        @logger.reopen(io)
-      else
-        @logger = create_logger(io)
-      end
+      @logger.reopen(io)
+    end
+
+    #
+    # Only log a warn message if it is not set to be ignored.
+    #
+    def warn(message, ids: [], &block)
+      msg = ids.empty? ? "" : "[#{ids.map(&:inspect).join(", ")}] "
+      msg += message
+      @logger.warn(msg, &block) unless @ignored.include?(:id)
```

Comment:
Probably don't want this to be based on `:id`. Should be `(@ignored & ids).any?` ?